### ISSUE TITLE
`confluence_cleanup_from_root` is missleading with `dry_run` enabled

### ISSUE THREAD

**Issue by Guillem96** [sphinx-contrib/confluencebuilder#847]
I understand that the default behviour when specifying `confluence_parent_page` will remove all pages not being deployed under the parent page.

I'm setting `confluence_cleanup_from_root` to `True` to change this behaviour and make the purge only happen under my index page.

For some reason when deploying with `confluence_publish_dryrun` enabled the logs show that some pages (located under the parent) are going to be deleted. 

The logs are missleading because they make you think that the cleanup is happening at parent level and not from the root level.

Version: 2.3.0
Python Version: 3.10

My sphinx configuration:

```
confluence_publish_token = os.getenv("CONFLUENCE_API_TOKEN")
confluence_ask_password = False
confluence_server_url = "...."
confluence_space_key = "...."
confluence_parent_page = 92209950
confluence_publish_prefix = "test_confluence_two - "
confluence_ignore_titlefix_on_index = True
confluence_ca_cert = os.getenv("CONFLUENCE_CA_BUNDLE")
confluence_disable_ssl_validation = True
confluence_cleanup_purge = True
confluence_cleanup_from_root = True
confluence_timeout = 20
confluence_publish_dryrun = (
    os.getenv("CONFLUENCE_PUBLISH_DRY_RUN", 'False').lower() in ('true', '1', 't')
)
```

Can you please check it?

**Comment by jdknight** [comment]
Thanks for reporting this issue. This should now be corrected with the changes included in #850. These changes will be made available next release (v2.4+), or can be tried out now using the [development installation](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/install/#development-installation).

**Comment by jdknight** [comment]
v2.4 is now available on PyPI -- marking as closed.
